torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -807,6 +928,12 @@ struct GraphFuser {
     for (auto it = block_->nodes().rbegin(); it != block_->nodes().rend();) {
       it = scanNodeForChunks(*it);
     }
+
+    // Remove outputs that have been added only because we need their size
+    for (Node * n : block_->nodes()) {
+      removeOutputsUsedOnlyInSize(n);
```

Comment:
nit: slightly misleading name because not only does it remove outputs only used in size, it also builds the shape expressions and adds them to the output

Reply:
Technically it doesn't "add the to the output", it just replaces `aten::size` calls on outputs with expressions computing sizes equal to those intermediates.